src/Cli/dotnet/NugetPackageDownloader/NuGetPackageDownloader.cs

```diff
@@ -157,18 +154,48 @@ private void VerifySigning(string nupkgPath)
                 return;
             }
 
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            if (!_isNuGetTool)
+            {
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    if (!_firstPartyNuGetPackageSigningVerifier.Verify(new FilePath(nupkgPath),
+                        out string commandOutput))
+                    {
+                        throw new NuGetPackageInstallerException(LocalizableStrings.FailedToValidatePackageSigning +
+                                                                 Environment.NewLine +
+                                                                 commandOutput);
+                    }
+                }
+            }
+            else
             {
-                if (!_firstPartyNuGetPackageSigningVerifier.Verify(new FilePath(nupkgPath),
-                    out string commandOutput))
+                if(RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
                 {
-                    throw new NuGetPackageInstallerException(LocalizableStrings.FailedToValidatePackageSigning +
-                                                             Environment.NewLine +
-                                                             commandOutput);
+                    if (!NuGetVerify(new FilePath(nupkgPath), out string commandOutput))
+                    {
+                        throw new NuGetPackageInstallerException(LocalizableStrings.FailedToValidatePackageSigning +
+                                                                 Environment.NewLine +
+                                                                 commandOutput);
+                    }
                 }
+                else
+                {
+                    _reporter.WriteLine(LocalizableStrings.NuGetPackageSignatureVerificationSkipped);
+                }   
             }
         }
 
+        private static bool NuGetVerify(FilePath nupkgToVerify, out string commandOutput)
+        {
+            var args = new[] { "verify", "--all", nupkgToVerify.Value };
+            var command = new DotNetCommandFactory(alwaysRunOutOfProc: true)
+                .Create("nuget", args);
+
+            var commandResult = command.CaptureStdOut().Execute();
+            commandOutput = commandResult.StdOut + Environment.NewLine + commandResult.StdErr;
+            return commandResult.ExitCode == 0;
+        }
+
```

Comment:
Can we leverage the existing logic instead of code duplication?https://github.com/dotnet/sdk/blob/45443375ec4a612d43b6888bb1c08770565614b8/src/Cli/dotnet/NugetPackageDownloader/FirstPartyNuGetPackageSigningVerifier.cs#L36